File: 226. Invert Binary Tree/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
  if (root) {
    const child = new TreeNode(root.val);
    child.left = invertTree(root.right);
    child.right = invertTree(root.left);
    return child;
  }
  return root;
};

File: 226. Invert Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if root:
            right, left = root.right, root.left
            root.left = self.invertTree(right)
            root.right = self.invertTree(left)
            return root
